cmd/master-controller-manager/controllers.go

```diff
@@ -85,6 +86,9 @@ func createAllControllers(ctrlCtx *controllerContext) error {
 	if err := masterconstrainttemplatecontroller.Add(ctrlCtx.ctx, ctrlCtx.mgr, ctrlCtx.log, 1, ctrlCtx.namespace, ctrlCtx.seedKubeconfigGetter); err != nil {
 		return fmt.Errorf("failed to create master constraint template controller: %v", err)
 	}
+	if err := masterconstraintsyncercontroller.Add(ctrlCtx.ctx, ctrlCtx.mgr, ctrlCtx.log, ctrlCtx.namespace, ctrlCtx.seedKubeconfigGetter); err != nil {
+		return fmt.Errorf("failed to create master constraint syncer controller: %v", err)
```

Comment:
Also, we actually have a controller to manage the seed lifecycles. I would suggest that you move your controller over there and use the seedsmapper instead of going through the seeds within you controller. Create a factory for your controller and add there: 
https://github.com/kubermatic/kubermatic/blob/master/cmd/master-controller-manager/controllers.go#L58

Reply:
good point, I also made a controller "master-constraint-template..." a while ago that could probably be moved to there in the future, and there is also the "userbinding" one which is similar.